### DISCUSSION TITLE
Recursive output.docker directories

### DISCUSSION THREAD

**Discussion by atomicbird** [srid/emanote#224] [Q&A] [answered]
**Describe the bug**

I'm running Emanote via the Docker image and finding that repeated runs lead to nested `output.docker` directories, even though I run the command in the same directory each time. Each level's content is a duplicate of the level above it.

**To Reproduce**

I'm getting it as follows:

1. `cd` to my Obsidian notebook directory
2. `mkdir output.docker` if it's not already there.
3. Run Emanote with

        docker run --platform linux/amd64 -e LANG=C.UTF-8 -e LC_ALL=C.UTF-8 -v $PWD:/data sridca/emanote emanote -L "/data" gen /data/output.docker

    I added the `--platform` because I'm running on Apple Silicon but otherwise this is the same as the suggested Docker invocation at https://emanote.srid.ca/start/install/docker.

The first time, the results go in `output.docker`. The second time they go in _both_ `output.docker/` and `output.docker/output.docker/`. Further runs lead to further duplicate nestings.

After a few updates it can get like this:

    find output.docker -name index.html -exec ls -l {} \;
    -rw-r--r--  1 tph  staff  18879 Jan  7 16:58 output.docker/index.html
    -rw-r--r--  1 tph  staff  18879 Jan  7 16:58 output.docker/output.docker/index.html
    -rw-r--r--  1 tph  staff  18879 Jan  7 16:58 output.docker/output.docker/output.docker/index.html
    -rw-r--r--  1 tph  staff  18879 Jan  7 16:58 output.docker/output.docker/output.docker/output.docker/index.html
    ....

Above, each `index.html` is identical. A new nested level appears every time.

**Desktop (please complete the following information):**

Docker Desktop 4.3.2 (72729) on macOS 12.1 (Monterey), Apple Silicon hardware.

**Comment by srid** [comment, marked as answer]
It is a bit of a hack in Docker workflow of emanote-template. This is because the output of first runs are fed as input to the second run, because that output is put inside source directory

The solution is to put the output directory **outside** of source directory. For that you have to mount the parent directory of source directory. I didn't bother doing this in GitHub Pages workflow, because `gen` is run only once.

**Comment by atomicbird** [comment]
Thanks, that did the trick.
